internal/http/services/owncloud/ocs/handlers/apps/sharing/shares/shares.go

```diff
@@ -463,6 +463,24 @@ func (h *Handler) extractPermissions(reqRole string, reqPermissions string, ri *
 			Error:   errors.New("cannot set the requested share permissions"),
 		}
 	}
+	// add a deny permission only if the user has the grant to deny (ResourcePermissions.DenyGrant == true)
+	if permissions == conversions.PermissionNone {
+		if !ri.PermissionSet.DenyGrant {
+			return nil, nil, &ocsError{
+				Code:    http.StatusNotFound,
+				Message: "Cannot set the requested share permissions: no deny grant on resource",
+				Error:   errors.New("Cannot set the requested share permissions: no deny grant on resource"),
+			}
+		}
+		existingPermissions := conversions.RoleFromResourcePermissions(ri.PermissionSet).OCSPermissions()
+		if permissions == conversions.PermissionInvalid || !existingPermissions.Contain(permissions) {
```

Comment:
Does this actually work? I mean `permissions == conversions.PermissionsNone` as stated in line 467. So

1) `permissions` can never be equal to `conversions.PermissionInvalid`
2) `existingPermissions` (which is `conversions.PermissionsAll` in best case) can NEVER contain `conversions.PermissionNone`.

So we would always run into the error below

Reply:
pushed a fix